core/arch/arm/kernel/boot.c

```diff
@@ -798,22 +799,43 @@ static int add_optee_dt_node(struct dt_descriptor *dt)
 		 * these.
 		 *
 		 * An SPI type of interrupt is indicated with a 0 in the
-		 * first cell.
+		 * first cell. A PPI type is indicated with value 1.
 		 *
 		 * The interrupt number goes in the second cell where
-		 * SPIs ranges from 0 to 987.
+		 * SPIs ranges from 0 to 987 and PPI ranges from 0 to 15.
 		 *
 		 * Flags are passed in the third cell where a 1 means edge
-		 * triggered.
+		 * triggered for SPI types and (8|mask<<8) means a lowlevel
+		 * triggered with CPU mask that apply for PPI types.
 		 */
-		const uint32_t gic_spi = 0;
-		const uint32_t irq_type_edge = 1;
-		uint32_t val[] = {
-			TEE_U32_TO_BIG_ENDIAN(gic_spi),
-			TEE_U32_TO_BIG_ENDIAN(CFG_CORE_ASYNC_NOTIF_GIC_INTID -
-					      GIC_SPI_BASE),
-			TEE_U32_TO_BIG_ENDIAN(irq_type_edge),
-		};
+		uint32_t val[3] = { };
+
+		/* PPI are visible only in current CPU cluster */
+		static_assert(!CFG_CORE_ASYNC_NOTIF_GIC_INTID ||
+			      CFG_CORE_ASYNC_NOTIF_GIC_INTID >= GIC_SPI_BASE ||
+			      (CFG_CORE_ASYNC_NOTIF_GIC_INTID >= GIC_PPI_BASE &&
+			       CFG_TEE_CORE_NB_CORE <= 4));
+
+		if (CFG_CORE_ASYNC_NOTIF_GIC_INTID >= GIC_SPI_BASE) {
+			/* GIC_SPI as per DT bindings */
+			val[0] = GIC_SPI;
```

Comment:
It would be more clear with separate helper variables for this, for instance, `itr_type`, `itr_id`, and `itr_trigger`.

Reply:
ok